neo4j-ogm-tests/neo4j-ogm-integration-tests/src/test/java/org/neo4j/ogm/domain/gh789/Entity.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2002-2020 "Neo4j,"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.neo4j.ogm.domain.gh789;
+
+import org.neo4j.ogm.annotation.Id;
+import org.neo4j.ogm.annotation.Index;
+import org.neo4j.ogm.annotation.NodeEntity;
+import org.neo4j.ogm.annotation.typeconversion.Convert;
+
+/**
+ * @author Michael J. Simons
+ */
+@NodeEntity
+public class Entity {
+
+    @Id
+    @Index(unique = true)
```

Comment:
The index annotation does not work as expected in combination with the auto-index manager and will stille produce statements like `CREATE CONSTRAINT ON (entity:Entity) ASSERT entity.key IS UNIQUE`.
This gives users a false idea of what happening. Of course also the `@Id` annotation would force the constraint/index creation but is needed to make this entity work.
My suggestion is to remove this line and we should point out somewhere that we cannot create index/constraints for composite attributes (does not matter if identifier or not) through Neo4j-OGM but this would be a DBA job.